HexChat/nowplaying.lua

```diff
@@ -91,42 +132,55 @@ local function format_timestamp (microsecs)
     return str
 end
 
-local function get_metadata(player, callback)
-    get_property(player, {'org.mpris.MediaPlayer2.Player', 'Metadata'}, callback) -- a{sv}
-end
-
-local function get_position(player, callback)
-    get_property(player, {'org.mpris.MediaPlayer2.Player', 'Position'}, callback) -- x
+local function template_string (template, replacements)
+    template = template:gsub("%$%$", "\000") -- $$ -> \000
+    template = template:gsub("%$([%w_]+)", replacements) -- $field
+    template = template:gsub("%${([^}]+)}", function (field) -- ${field}; ${field=default}
+        field, default = field:match('(.+)=(.*)') or field
+        repl = replacements[field]
+        if repl == nil and repl ~= nil then
```

Comment:
This also doesn't make sense.

(Taking notes for myself when I review this PR since I use Linux now but also foobar2000, which doesn't support mpris.)